src/ios/registerNativeModule.js

```diff
@@ -50,6 +54,73 @@ const addProjectToLibraries = (libraries, file) => {
 };
 
 /**
+ * Given folder, it returns an array of all header files
+ * inside it, ignoring node_modules and examples
+ */
+const getHeadersInFolder = (folder) =>
+  glob
+    .sync('**/*.h', {
+      cwd: folder,
+      nodir: true,
+      ignore: GLOB_EXCLUDE_PATTERN,
+    })
+    .map(file => path.join(folder, file));
+
+/**
+ * Given an array of headers it returns search path so Xcode can find them
+ * If there are multiple header files located accross different folders,
+ * it finds the top-most one and returns recursive version of it
+ */
+const getHeaderSearchPath = (sourceDir, headers) => {
+  const directories = union(
+    headers.map(path.dirname)
+  );
+
+  if (directories.length === 1) {
+    return `"$(SRCROOT)/${path.relative(sourceDir, directories[0])}"`;
+  }
+
+  const directory = directories.reduce((topMostDir, currentDir) => {
+    return currentDir.split(path.sep) < topMostDir.split(path.sep)
+      ? currentDir
+      : topMostDir;
+  });
+
+  return `"$(SRCROOT)/${path.relative(sourceDir, directory)}/**"`;
+};
+
+
+/**
+ * Given Xcode project and path, iterate over all build configurations
+ * and append new path to HEADER_SEARCH_PATHS section
+ *
+ * We cannot use builtin addToHeaderSearchPaths method since react-native init does not
+ * use $(TARGET_NAME) for PRODUCT_NAME, but sets it manually so that method will skip
+ * that target.
+ *
+ * To workaround that issue and make it more bullet-proof for different names,
+ * we iterate over all configurations and look if React is already there. If it is,
+ * we assume user will need newly added dependency headers in it as well.
+ */
+const addToHeaderSearchPaths = (project, path) => {
+  const config = project.pbxXCBuildConfigurationSection();
+
+  Object
+    .keys(config)
+    .filter(ref => ref.indexOf('_comment') === -1)
```

Comment:
node-xcode stores _comment for targets that describe whether its Debug or Release. We do not need that information and that's also what they do when iterating.